# Exploring US Gun Deaths Data

### A Sample of the first 5 rows of Gun Deaths Data

In [9]:
import csv #Import CSV module
f = open("guns.csv", 'r') #Open the csv file using open function 'r' for reading
data = list(csv.reader(f)) #Nests the CSV reader function on the Guns Data file in a list function to convert to list
print(data[:5]) #Print First Five Rows to Verify

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


### Column Headers of Data

In [10]:
header = data[0] #Extracts the header column from the data which is at index 0
print(header) #Prints header

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']


### Cleaning the Data to Remove Column Headers

In [11]:
data = data[1:] #Removes the Header column
print(data[0:5]) #Verify header column is removed

[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


### Counting Gun Deaths by Year

In [12]:
years = [row[1] for row in data] #List comprehension that extracts year data from each row at index 1
year_counts = {} #Create a dictionary year_counts

#Populating the year_counts dictionary with a for loop
for year in years: #Loops through the years list
    if year in year_counts: #Checks to see if the year is in the year_counts dictionary
        year_counts[year] += 1 #If it is in the dictionary it will add to its value by 1
    else:
        year_counts[year] = 1 #If it is not in the dictionary it will set its value to 1

print(year_counts)

{'2014': 33599, '2013': 33636, '2012': 33563}


Judging by the count of gun deaths by year, 2014 - 33599, 2013 - 33636, 2012 - 33563, the number of gun deaths hasn't drastically changed over the three years.

### Exploring Gun Deaths by Month and Year

In [13]:
import datetime #Import the datetime module
dates = [datetime.datetime(year=int(row[1]), month=int(row[2]), day=1) for row in data] #Use a list comprehension to extract date time objects using the datetime function
dates[0:5] #Verify the dates are correct

[datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0)]

In [14]:
date_counts = {} #Create an empty dictionary for date counts
for date in dates: #Loops through the dates objects list
    if date in date_counts: #Checks to see if the date is in the date counts dictionary
        date_counts[date] += 1 #If the date is already there it will add to its value by 1
    else:
        date_counts[date] = 1 #if the date is not in the dictionary it will set its value to 1

date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

### Exploring Gun Deaths by Race and Sex

In [15]:
sex_counts = {}
sexes = [row[5] for row in data]

for sex in sexes:
    if sex in sex_counts:
        sex_counts[sex] += 1
    else:
        sex_counts[sex] = 1

sex_counts

{'F': 14449, 'M': 86349}

The number of gun deaths is much higher for Males than it is for Females, almost 6 times as much. 

In [16]:
race_counts = {}
races = [row[7] for row in data]

for race in races:
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1
race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

Persons of white race have the highest number of gun deaths but that is expected due to whites being the majority, there is likely a higher % of gun deaths for minorities based on the number of gun deaths and an estimation of population % of minorities.

### Using Census Data for further exploration of Gun Deaths data

In [17]:
import csv
y = open("census.csv", 'r')
censusdata = list(csv.reader(y))
print(censusdata)
censusdata

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


[['Id',
  'Year',
  'Id',
  'Sex',
  'Id',
  'Hispanic Origin',
  'Id',
  'Id2',
  'Geography',
  'Total',
  'Race Alone - White',
  'Race Alone - Hispanic',
  'Race Alone - Black or African American',
  'Race Alone - American Indian and Alaska Native',
  'Race Alone - Asian',
  'Race Alone - Native Hawaiian and Other Pacific Islander',
  'Two or More Races'],
 ['cen42010',
  'April 1, 2010 Census',
  'totsex',
  'Both Sexes',
  'tothisp',
  'Total',
  '0100000US',
  '',
  'United States',
  '308745538',
  '197318956',
  '44618105',
  '40250635',
  '3739506',
  '15159516',
  '674625',
  '6984195']]

As the race names in the two data sets are different we need to create two different dictionaries to calculate the gun deaths by race per capita. 

# NEEDS TO BE CHECKED >>

In [18]:
mapping = {
    "Asian/Pacific Islander": 15159516 + 674625,
    "Native American/Native Alaskan": 3739506,
    "Black": 40250635,
    "Hispanic": 44618105,
    "White": 197318956
}

race_per_hundredk = {}
for k,v in race_counts.items():
    race_per_hundredk[k] = (v / mapping[k]) * 100000

race_per_hundredk

{'Asian/Pacific Islander': 0,
 'Black': 0,
 'Hispanic': 0,
 'Native American/Native Alaskan': 0,
 'White': 0}

### Exploring Gun Deaths by Intent

In [23]:
mapping

{'Asian/Pacific Islander': 15834141,
 'Black': 40250635,
 'Hispanic': 44618105,
 'Native American/Native Alaskan': 3739506,
 'White': 197318956}

In [21]:
intents = [row[3] for row in data]
homicide_race_counts = {}

for i,race in enumerate(races):
    if race not in homicide_race_counts:
        homicide_race_counts[race] = 0
    if intents[i] == "Homicide":
        homicide_race_counts[race] += 1

homicide_race_counts

{'Asian/Pacific Islander': 559,
 'Black': 19510,
 'Hispanic': 5634,
 'Native American/Native Alaskan': 326,
 'White': 9147}

In [28]:
race_per_hundredk = {}
for k,v in homicide_race_counts.items():
    race_per_hundredk[k] = (float(v) / mapping[k]) * 100000

race_per_hundredk

{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}